client/post-editor/edit-post-status/index.jsx

```diff
@@ -195,16 +195,15 @@ class EditPostStatus extends Component {
 			return;
 		}
 
-		const { status, password, type } = this.props.post || {};
+		const { password, type } = this.props.post || {};
 		const isPrivateSite = this.props.site && this.props.site.is_private;
 		const savedStatus = this.props.savedPost ? this.props.savedPost.status : null;
 		const savedPassword = this.props.savedPost ? this.props.savedPost.password : null;
 		const props = {
-			visibility: utils.getVisibility( this.props.post ),
+			status: this.props.status,
```

Comment:
Did the logic need to be moved up as a prop? Seems like it would have been enough to simply remove the `utils.getVisibility`.